README.md

```diff
@@ -43,3 +43,48 @@ It seems that sendResponse is not supported by webextension-polyfill:
 [https://github.com/mozilla/webextension-polyfill/issues/16/#issuecomment-296693219]
 The reason seems to be that a response callback might be removed from `onMessage` some time later. 
 Because of that, we have to implement our own request-response matching mechanism with `MessagingService`.
+
+## Default parameters
+
+### Webextension state (active or inactive)
+Default webextension state is "inactive".
+
+### Panel state (open or closed)
+Default panel state is determined by a setting in configuration. If not configured by user explicitly,
+it is "open".
+
+### Selected panel tab
+Default panel tab is "info".
+
+## Usage Scenarios
+
+Webextension uses a tab object to start state of an extension within a certain browser tab or window.
+Because of this, each tab has its own state that is completely isolated. All secenarios listed below
+will describe behavior of webextension within a single tab.
+
+### Activate for the first time in a tab
+When extension is activated in a tab for the first time, it uses default parameters to set its state.
+
+### Activated, navigate to the new page
+When an extension is activated, and user navigate to the other page, the webextension retains its
+active state. All other state parameters (panel state and selected tab) are set to their default values.
```

Comment:
Actually, this is not quite right. What we wanted was for when we navigate to a new page, for the panel and tab to retain their state IF the extension was activated. This was finally working in a prior commit but these changes have reversed it :-(

Reply:
OK, I've updated it.